src/main/java/org/jenkinsci/plugins/runselector/selectors/BuildNumberRunSelector.java

```diff
@@ -0,0 +1,52 @@
+package org.jenkinsci.plugins.runselector.selectors;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.Run;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.runselector.RunSelectorDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+
+/**
+ * Select the run based on its build number.
+ *
+ * @author Alexandru Somai
+ */
+public class BuildNumberRunSelector extends AbstractSpecificRunSelector {
+
+    @DataBoundConstructor
+    public BuildNumberRunSelector(String buildNumber) {
```

Comment:
It's a String, because it accepts build parameters too (useful in Freestyle project)